agon_audio.h

```diff
@@ -8,20 +8,59 @@
 
 #pragma once
 
+extern void sendChannelStatus(int channel, int status);
+
+#define SAMPLE_SLOTS 6
+
+struct audio_sample
+{
+    
+    bool written = false;         // has this sample been written yet? If note, ignore efforts to play.
+    uint16_t length;              // sample length
+    uint16_t rate;                // sample rate
+    int8_t sample_buffer[262144]; // Max 256KB of raw PCM
+
+};
+
+struct audio_sample* samples[SAMPLE_SLOTS]; //6 Samples = ~1.5MB of PSRAM

```

Comment:
I kinda liked your simplistic sample storage... but it's obviously rather restrictive with its fixed buffers, and is rather limited.

how to best handle loading and storing samples was my biggest headache - and indeed forced me to learn quite a lot about dynamic memory management in C++.

possibly the most frustrating aspect of dealing with samples tho is how slow BASIC can be reading in a file and sending it over to the VDP. 😁 my test code used a 160kb sample file, which took about 90s to send across.  oof!

(I think with appropriate z80 assembly code that transfer could be cut down to about 2s)